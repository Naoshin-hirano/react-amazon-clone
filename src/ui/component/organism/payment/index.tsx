import React from "react";
import "../../../../design/component/atoms/payment/index.css";
import { Link, useLocation } from "@tanstack/react-location";
import { CardElement } from "@stripe/react-stripe-js";
import CurrencyFormat from "react-currency-format";
import { CheckoutProduct } from "../common/checkout-product";

// 決済には@stripe/react-stripe-jsを使用
export const Payment: React.FC<any> = (props) => {
    const { basket } = props;
    const { history } = useLocation();

    const handleSubmit = () => {
        // 支払い処理 → 支払い完了後に/orders画面へ遷移
        console.log("支払い処理");
        history.push("/orders");
    };

    const handleChange = () => {};
    return (
        <div className="payment">
            <div className="payment__container">
                <h1>
                    Checkout (<Link to="/checkout">{basket?.length} items</Link>
                    )
                </h1>
                <div className="payment__section">
                    <div className="payment__title">
                        <h3>Delivery Address</h3>
                    </div>
                    <div className="payment__address">
                        <p>xxxx@yyyy.com</p>
                        <p>123 React Lane</p>
                        <p>Los Angeles, CA</p>
                    </div>
                </div>
                <div className="payment__section">
                    <div className="payment__title">
                        <h3>Review Items and Delivery</h3>
                    </div>
                    <div className="payment__items">
                        {basket.map((item: any, key: any) => (
                            <CheckoutProduct
                                key={key}
                                id={item.id}
                                title={item.title}
                                image={item.image}
                                price={item.price}
                                rating={item.rating}
                            />
                        ))}
                    </div>
                </div>

                <div className="payment__section">
                    <div className="payment__title">
                        <h3>Payment Method</h3>
                    </div>
                    <div className="payment__details">
                        <form onSubmit={handleSubmit}>
                            <CardElement onChange={handleChange} />

                            <div className="payment__priceContainer">
                                <CurrencyFormat
                                    renderText={(value) => (
                                        <h3>Order Total: {value}</h3>
                                    )}
                                    decimalScale={2}
                                    value="77,555"
                                    displayType={"text"}
                                    thousandSeparator={true}
                                    prefix={"$"}
                                />
                                {/* 処理中または処理完了時は非活性 */}
                                <button>
                                    <span>Buy Now</span>
                                </button>
                            </div>

                            {/* {error && <div>{error}</div>} */}
                        </form>
                    </div>
                </div>
            </div>
        </div>
    );
};
