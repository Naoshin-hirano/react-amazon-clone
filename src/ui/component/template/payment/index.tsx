import React from "react";
import { Header } from "ui/component/organism/common/header";
import { Payment } from "ui/component/organism/payment";
import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";

const promise = loadStripe(
    "pk_test_51GxK2EFjdfjgcidOJRFut1KmR2wihbdrukB6yshUqRlP1QtTna5ZRjYTGcgFgg1wwvVf5c4vNp3mw8FCRVq3dUnW0034b3GyjS"
);

export const Template: React.FC<any> = (props) => {
    return (
        <div>
            <Header />
            <Elements stripe={promise}>
                <Payment {...props} />
            </Elements>
        </div>
    );
};
