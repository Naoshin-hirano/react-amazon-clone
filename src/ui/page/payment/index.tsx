import React from "react";
import { Template } from "ui/component/template/payment";
import { useCreateProps } from "./presentation";

export const PaymentPage: React.FC = () => {
    const props = useCreateProps();
    return <Template {...props}></Template>;
};
