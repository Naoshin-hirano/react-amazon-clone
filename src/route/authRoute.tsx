import { Route } from "@tanstack/react-location";
import React from "react";
import { LoginPage } from "ui/page/login";
import { RegisterPage } from "ui/page/register";

export const authRoute: Route[] = [
    {
        path: "/login",
        element: <LoginPage />,
    },
    {
        path: "/register",
        element: <RegisterPage />,
    },
];
