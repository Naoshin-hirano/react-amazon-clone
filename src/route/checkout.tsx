import { Route } from "@tanstack/react-location";
import React from "react";
// import { SocondSamplePage } from "ui/page/second";
import { CheckoutPage } from "ui/page/checkout";

export const checkoutRoute: Route[] = [
    {
        path: "/checkout",
        element: <CheckoutPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
