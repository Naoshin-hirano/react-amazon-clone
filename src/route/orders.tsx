import { Route } from "@tanstack/react-location";
import React from "react";
import { OrdersPage } from "ui/page/orders";

export const ordersRoute: Route[] = [
    {
        path: "/orders",
        element: <OrdersPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
