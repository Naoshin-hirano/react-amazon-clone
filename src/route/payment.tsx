import { Route } from "@tanstack/react-location";
import React from "react";
import { PaymentPage } from "ui/page/payment";

export const paymentRoute: Route[] = [
    {
        path: "/payment",
        element: <PaymentPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
