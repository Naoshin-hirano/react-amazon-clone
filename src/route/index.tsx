import { ReactLocation, Route } from "@tanstack/react-location";
import { history } from "./history";
import { authRoute } from "route/authRoute";
import React from "react";
import { TopPage } from "ui/page/top";

const indexRoute: Route[] = [
    {
        path: "/",
        element: <TopPage />,
    },
    {
        path: "/top",
        element: <TopPage />,
    },
];

export const routes: Route[] = [authRoute].reduce(
    (prevRoute, currRoute) => prevRoute.concat(currRoute),
    indexRoute
);

export const location = new ReactLocation({ history: history });
