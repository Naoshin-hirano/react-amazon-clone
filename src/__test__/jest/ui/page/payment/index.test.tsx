import { render } from "@testing-library/react";
import React from "react";
import { Router } from "@tanstack/react-location";
import { location, routes } from "../../../../../route";
import { PaymentPage } from "ui/page/payment";

describe("PaymentPage", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("PaymentPage画面 snapshot確認", async () => {
        const { asFragment } = render(
            <Router routes={routes} location={location}>
                <PaymentPage />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot("PaymentPage");
    });
});
