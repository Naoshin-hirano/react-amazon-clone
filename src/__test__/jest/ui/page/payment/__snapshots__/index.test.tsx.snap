// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentPage PaymentPageÁîªÈù¢ snapshotÁ¢∫Ë™ç: PaymentPage 1`] = `
<DocumentFragment>
  <div>
    <div
      class="header"
    >
      <a
        class="active"
        href="/"
      >
        <img
          alt="Amazon logo"
          class="header__logo"
          src="http://pngimg.com/uploads/amazon/amazon_PNG11.png"
        />
      </a>
      <div
        class="header__search"
      >
        <input
          class="header__searchInput"
          type="text"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root header__searchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <div
        class="header__nav"
      >
        <a
          href="/login"
        >
          <div
            class="header__option"
          >
            <span
              class="header__optionLineOne"
            >
              Âπ≥ÈáéÁõ¥Âº•„Åï„Çì
            </span>
            <span
              class="header__optionLineTwo"
            >
              „É≠„Ç∞„Ç§„É≥
            </span>
          </div>
        </a>
        <a
          href="/orders"
        >
          <div
            class="header__option"
          >
            <span
              class="header__optionLineOne"
            >
              ËøîÂìÅ„ÇÇ„Åì„Å°„Çâ
            </span>
            <span
              class="header__optionLineTwo"
            >
              Ê≥®ÊñáÂ±•Ê≠¥
            </span>
          </div>
        </a>
        <div
          class="header__option"
        >
          <span
            class="header__optionLineOne"
          >
            „ÅÇ„Å™„Åü„ÅÆ
          </span>
          <span
            class="header__optionLineTwo"
          >
            „Ç¢„Ç´„Ç¶„É≥„Éà
          </span>
        </div>
        <a
          href="/checkout"
        >
          <div
            class="header__optionBasket"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.21 9l-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
            <span
              class="header__optionLineTwo header__basketCount"
            >
              3
            </span>
          </div>
        </a>
      </div>
    </div>
    <div
      class="payment"
    >
      <div
        class="payment__container"
      >
        <h1>
          Checkout (
          <a
            href="/checkout"
          >
            3 items
          </a>
          )
        </h1>
        <div
          class="payment__section"
        >
          <div
            class="payment__title"
          >
            <h3>
              Delivery Address
            </h3>
          </div>
          <div
            class="payment__address"
          >
            <p>
              xxxx@yyyy.com
            </p>
            <p>
              123 React Lane
            </p>
            <p>
              Los Angeles, CA
            </p>
          </div>
        </div>
        <div
          class="payment__section"
        >
          <div
            class="payment__title"
          >
            <h3>
              Review Items and Delivery
            </h3>
          </div>
          <div
            class="payment__items"
          >
            <div
              class="checkoutProduct"
            >
              <img
                alt="checkoutProductImg"
                class="checkoutProduct__image"
                src="https://www.anoyama.com/wp-content/uploads/2019/05/0_equipment-860x550.jpg"
              />
              <div
                class="checkoutProduct__info"
              >
                <p
                  class="checkoutProduct__title"
                >
                  ÁôªÂ±±„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
                </p>
                <p
                  class="checkoutProduct__price"
                >
                  <small>
                    ¬•
                  </small>
                  <strong>
                    23,400
                  </strong>
                </p>
                <div
                  class="checkoutProduct__rating"
                >
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                </div>
                <button>
                  „Ç´„Éº„Éà„Åã„ÇâÂâäÈô§
                </button>
              </div>
            </div>
            <div
              class="checkoutProduct"
            >
              <img
                alt="checkoutProductImg"
                class="checkoutProduct__image"
                src="https://images.camphack.jp/wp-content/uploads/2022/02/image19.jpeg"
              />
              <div
                class="checkoutProduct__info"
              >
                <p
                  class="checkoutProduct__title"
                >
                  „Ç≠„É£„É≥„Éó„Ç∞„ÉÉ„Ç∫‰∏ÄÂºè
                </p>
                <p
                  class="checkoutProduct__price"
                >
                  <small>
                    ¬•
                  </small>
                  <strong>
                    53,400
                  </strong>
                </p>
                <div
                  class="checkoutProduct__rating"
                >
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                </div>
                <button>
                  „Ç´„Éº„Éà„Åã„ÇâÂâäÈô§
                </button>
              </div>
            </div>
            <div
              class="checkoutProduct"
            >
              <img
                alt="checkoutProductImg"
                class="checkoutProduct__image"
                src="https://m.media-amazon.com/images/I/616AAPLKJTL._AC_UY500_.jpg"
              />
              <div
                class="checkoutProduct__info"
              >
                <p
                  class="checkoutProduct__title"
                >
                  ÁôªÂ±±„Ç∑„É•„Éº„Ç∫
                </p>
                <p
                  class="checkoutProduct__price"
                >
                  <small>
                    ¬•
                  </small>
                  <strong>
                    16,400
                  </strong>
                </p>
                <div
                  class="checkoutProduct__rating"
                >
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                  <p>
                    üåü
                  </p>
                </div>
                <button>
                  „Ç´„Éº„Éà„Åã„ÇâÂâäÈô§
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="payment__section"
        >
          <div
            class="payment__title"
          >
            <h3>
              ÊîØÊâï„ÅÑÊñπÊ≥ï
            </h3>
          </div>
          <div
            class="payment__details"
          >
            <form>
              <div />
              <div
                class="payment__priceContainer"
              >
                <h3>
                  ÂêàË®à: $77,555
                </h3>
                <button>
                  <span>
                    Ë≥ºÂÖ•„Åô„Çã
                  </span>
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
